### DISCUSSION TITLE
Thank you for creating GitFiend

### DISCUSSION THREAD

**Discussion by nav9** [GitFiend/Support#150] [General]
I just started using GitFiend, and am already happy to see how well various parts of the interface are designed. Easy and intuitive to use. You've really put some thought into how people use Git. Thank you for creating it.

**Comment by GitFiend** [comment]
Thanks @nav9! That's high praise.

**Comment by nav9** [comment]
You are most welcome. As I've used GitFiend, I noticed that merging is one functionality I use frequently, and it'd help to have a convenient way to do it. I also have a hunch that y'all have a plan to consider and implement such things based on the user journey. All the best!

**Reply by GitFiend** [reply]
There are plans to improve the branches view (just released a new version today with some improvements here), but it would definitely be helpful to get a fresh perspective on what parts of merging you find inconvenient.

**Comment by nav9** [comment]
Ok, firstly, the compliment I gave was genuine. It wasn't a way to sneak in suggestions. I tried a lot of other (fully) free Git clients for Linux, and GitFiend was indeed the one I felt most comfortable with.
Since you asked, here are some points:
1. When I open GitFiend, the "commits" section is shown first. However, what I open GitFiend most often for, is to commit my changes. So it'd help to show the "Changes" section by default.
![image](https://user-images.githubusercontent.com/2093933/218259277-5f654d43-eea6-4019-9c0e-8411cc09e389.png)
2. It takes a while to realize that the three dots icon is a menu icon. The three horizontal lines (hamburger icon) would be more noticeable. 
![image](https://user-images.githubusercontent.com/2093933/218259396-db34a921-b91b-45e9-80a9-1f6fccda5cf9.png)
3. The merge functionality could be provided from the "Commits" view (clicking on a branch name) or the "Branches" view or even as an icon next to the "Fetch" button. So far I don't see any option to initiate a merge. One really un-intuitive aspect of Git is that to merge branch1 into branch2, one is supposed to first checkout branch2. It'd be really convenient if GitFiend provided the option for a person to click any branch they wanted, and merge it into any branch they specify or specify a branch they want to merge into the selected branch. GitFiend can automatically do the checkout of the appropriate branch (or provide a warning of uncommitted changes, and perhaps offer the option to auto-commit with an auto-generated message).
4. Branches are better visualized like branches of a tree, rather than like horizontal strips. 
![image](https://user-images.githubusercontent.com/2093933/218259891-7d6e05f2-9118-423e-9f17-c64d9857eaa7.png)
5. When the "Changes" section is selected, it helps to auto-set the cursor into the textfield where the commit message is to be typed. 
6. It helps to have tooltips on mouse hover. For example, I initially didn't understand what the checkboxes are for, in the "changes" section. 
![image](https://user-images.githubusercontent.com/2093933/218259966-ff0c6879-4e06-499f-82f3-44bd3c4e1184.png)
Hope this helps.

**Comment by GitFiend** [comment]
Thanks for the feedback. I think I agree with most of it, and I've already made a few changes. I recently tried to imagine how 4 would work, but chose to make a smaller change in the short term as I saw too many issues in the case of say 1000 branches. I've definitely seen people be confused by 3. It's worth thinking about more.
